internal/cli/kraft/cloud/deploy/deploy.go

```diff
@@ -0,0 +1,304 @@
+	// performing a type of build.
+	for _, candidate := range deployers {
+		log.G(ctx).
+			WithField("packager", candidate.String()).
+			Trace("checking compatibility")
+
+		capable, err := candidate.Deployable(ctx, opts, args...)
+		if capable && err == nil {
+			d = candidate
+			break
+		}
+
+		log.G(ctx).
+			WithError(err).
+			WithField("packager", candidate.String()).
+			Trace("incompatbile")
+	}
+
+	if d == nil {
+		return fmt.Errorf("could not determine what or how to deploy from the given context")
+	}
+
+	packs, err := d.Prepare(ctx, opts, args...)
+	if err != nil {
+		return fmt.Errorf("could not prepare deployment: %w", err)
+	}
+
+	// FIXME(nderjung): Gathering the digest like this really dirty.
+	metadata := packs[0].Columns()
+	var digest string
+	for _, m := range metadata {
+		if m.Name != "index" {
+			continue
+		}
+
+		digest = m.Value
+	}
+
+	// TODO(nderjung): This is a quirk that will be removed.  Remove the `index.`
+	// from the name.
+	if opts.Name[0:17] == "index.unikraft.io" {
+		opts.Name = opts.Name[6:]
+	}
+
+	paramodel, err := processtree.NewProcessTree(
+		ctx,
+		[]processtree.ProcessTreeOption{
+			processtree.IsParallel(false),
+			processtree.WithRenderer(
+				log.LoggerTypeFromString(config.G[config.KraftKit](ctx).Log.Type) != log.FANCY,
+			),
+			processtree.WithFailFast(true),
+			processtree.WithHideOnSuccess(true),
+			processtree.WithTimeout(opts.Timeout),
+		},
+		processtree.NewProcessTreeItem(
+			"deploying",
+			"",
+			func(ctx context.Context) error {
+			checkRemoteImages:
+				for {
+					images, err := opts.Client.Images().WithMetro(opts.Metro).List(ctx, nil)
+					if err != nil {
+						return fmt.Errorf("could not check list of images: %w", err)
+					}
+
+					for _, image := range images {
+						split := strings.Split(image.Digest, "/")
+						if !strings.HasPrefix(split[len(split)-1], digest) {
+							continue
+						}
+
+						break checkRemoteImages
+					}
+				}
+
+				time.Sleep(time.Second)
+
+				return nil
+			},
+		),
+	)
+	if err != nil {
+		return err
+	}
+
+	if err := paramodel.Start(); err != nil {
+		return err
+	}
+
+	instance, err := create.Create(ctx, &create.CreateOptions{
+		Env:      opts.Env,
+		Memory:   opts.Memory,
+		Ports:    opts.Ports,
+		Replicas: opts.Replicas,
+		Start:    !opts.NoStart,
+		Name:     opts.Name,
+		Metro:    opts.Metro,
+	}, append([]string{opts.Name}, args...)...)
+	if err != nil {
+		return fmt.Errorf("could not create instance: %w", err)
+	}
+
+	log.G(ctx).
+		WithField("uuid", instance.UUID).
+		Debug("created instance")
+
+	for {
+		instance, err = opts.Client.Instances().WithMetro(opts.Metro).Status(ctx, instance.UUID)
+		if err != nil {
+			return fmt.Errorf("could not get instance status: %w", err)
+		}
+
+		if instance.Status == "starting" {
+			continue
+		}
+
+		break
+	}
+
+	dns := instance.DNS
+	if len(dns) > 0 {
+		for _, port := range opts.Ports {
+			if strings.HasPrefix(port, "443") {
+				dns = "https://" + dns
```

Comment:
You should check if it already has a prefix

Reply:
The API does not return a URI scheme